### DISCUSSION TITLE
Stuggling to get WCA on rails to run

### DISCUSSION THREAD

**Discussion by elfishhenry** [thewca/worldcubeassociation.org#9418] [Q&A] [answered]
![image](https://github.com/thewca/worldcubeassociation.org/assets/109970614/fadebad1-5641-40bc-8c98-83392d8467cb)
The picture shows the only feed back I receive from attempting to "docker-compose up" it, how do I fix?

**Comment by gregorbg** [comment]
Hey,

sorry for the late reply. We are not used to handling support inquiries via this "GitHub Discussions" feature.

The screenshot looks like you have downloaded the code to a Windows machine, but are trying to run inside a Unix Docker host. The `\r` at the end is a typical symptom of incorrectly converting line endings between Unix and Windows: https://www.cs.toronto.edu/~krueger/csc209h/tut/line-endings.html

**Reply by gregorbg** [reply, marked as answer]
To resolve this issue, you need to consider your own development environment (which OS are you using to download the code from GitHub, where is it stored, which OS runner is being used to exec Docker?) and look into line ending conversions when doing `git clone`. 

ChatGPT can also help a lot with these kinds of "classical" problems. Prompts like "I am working on a Windows machine and have downloaded code from GitHub. There is a Docker enviroment which uses a Unix-based container in a Docker Compose setup, and the command defined in `docker-compose.yml` throws an error because of mis-converted line-breaks. How to fix this issue?" can help.
